Updated Configuration Settings (#5239)
mattermost/docs: source/configure/configuration-settings.rst

1. Find the setting in ``config.json``. In this case, *ServiceSettings.SiteURL*.
2. Add ``MM_`` to the beginning and convert all characters to uppercase and replace the ``.`` with ``_``. For example, *MM_SERVICESETTINGS_SITEURL*.

> 3. The setting becomes ``export MM_SERVICESETTINGS_SITEURL="http://example.com"``. An environment variable (ENV) file is created under `/etc/mattermost`.

Reviewer: > An environment variable (ENV) file is created under `/etc/mattermost`.

Can you clarify this line? I think the admin needs to manually create this file and this is only needed when Mattermost is deployed via an init file, and the entry also needs to be added in the init file. Basically, it can be any file that the user wishes.

Did you mean to say all of that? Or something else?

Author: I could use your help clarifying this information, @agnivade. This docs request is coming from a [Community-based conversation](https://community-daily.mattermost.com/core/pl/hdme69zqwjgh5qaurc4oqsbexw). Would you be open to drafting an update?

Reviewer: I took a look at the thread. It seems the user is not the person who configured the system.

Basically there are 2 things here:
1. A config setting configured via an env var cannot be modified from the system console. This is already documented.
2. If Mattermost is run from an init file, the environment variables can either be set via `Environment=<>`, or `EnvironmentFile=<path/to/file>`. In the second case, the file should contain the list of environment variables to set. More info here: https://www.man7.org/linux/man-pages/man5/systemd.exec.5.html#ENVIRONMENT

I think 1 is already documented. I'll leave it to you to see how/where you want to incorporate 2.

Author: Thank you so much for your help with this, @agnivade.

2. Add ``MM_`` to the beginning and convert all characters to uppercase and replace the ``.`` with ``_``. For example, *MM_SERVICESETTINGS_SITEURL*.
3. The setting becomes ``export MM_SERVICESETTINGS_SITEURL="http://example.com"``.

.. note::

  - If Mattermost is run from an initialization file, environment variables can be set via ``Environment=<>``, or ``EnvironmentFile=<path/to/file>``. In the second case, the file specified contains the list of environment variables to set. 
  - When settings are configured through an environment variable, System Admins can't modify them in the System Console. If a setting is configured through an environment variable, and any other changes are made in the System Console, the value stored of the environment variable will be written back to the ``config.json`` as that setting's value.

>   - For any setting that is not set in ``config.json`` or in environment variables, the Mattermost server uses the setting's default value as documented in the sections below on this page.

Reviewer 2: ```suggestion
  - For any setting that's not set in ``config.json`` or in environment variables, the Mattermost server uses the setting's default value as documented in the sections below on this page.
```

**This field is required in Mattermost v3.8 and later.**

> In Mattermost v5.1 and later, the URL may contain a subpath, such as ``"https://example.com/company/mattermost"``. System Admins can't update the Site URL value when it's configured through an environment variable. See the `Environment Variables <https://docs.mattermost.com/configure/configuration-settings.html#environment-variables>`__ product documentation for details. 

Reviewer 2: ```suggestion
In Mattermost v5.1 and later, the URL may contain a subpath, such as ``"https://example.com/company/mattermost"``. System Admins can't update the Site URL value when it's configured through an environment variable. See the `Environment Variables <https://docs.mattermost.com/configure/configuration-settings.html#environment-variables>`__ product documentation for details.
```

> If Site URL is not set, the following features will not operate correctly:

Reviewer 2: ```suggestion
If Site URL isn't set, the following features will not operate correctly:
```

**This field is required in Mattermost v3.8 and later.**

> In Mattermost v5.1 and later, the URL may contain a subpath, such as ``"https://example.com/company/mattermost"``. System Admins can't update the Site URL value if an environment variable file exists under `/etc/mattermost`. See the `Environment Variables <https://docs.mattermost.com/configure/configuration-settings.html#environment-variables>`__ documentation for more information on working with environment variables. 

Reviewer: Same question as above.
